client/mysqldump.c

```diff
@@ -5874,6 +5918,10 @@ static char *primary_key_fields(const char *table_name)
     {
       quoted_field= quote_name(row[4], buff, 0);
       result_length+= strlen(quoted_field) + 1; /* + 1 for ',' or \0 */
+      if (desc)
+      {
+        result_length+= 5; /* + 1 for space, and + 4 for "DESC" */
```

Comment:
I'd do static const char *desc_index= " DESC"; here strlen, below use the var directly, but this is strictly up to you